common/flatpak-context.c

```diff
@@ -852,11 +916,54 @@ flatpak_context_parse_filesystem (const char             *filesystem_and_mode,
   return FALSE;
 }
 
+/* Note: This only works with valid keys, i.e. they passed flatpak_context_parse_filesystem */
+static gboolean
+flatpak_filesystem_key_in_home (const char *filesystem)
+{
+  /* "home" is definitely in home */
+  if (strcmp (filesystem, "home") == 0)
+    return TRUE;
+
+  /* All the other special fs:es are non-home.
+   * Note: This considers absolute paths that are in the homedir as non-home.
+   */
+  if (g_strv_contains (flatpak_context_special_filesystems, filesystem) ||
+      g_str_has_prefix (filesystem, "/"))
+    return FALSE;
+
+  /* Files in xdg-run are not in home */
+  if (g_str_has_prefix (filesystem, "xdg-run"))
+    return FALSE;
+
+  /* All remaining keys (~/, xdg-data, etc) are considered in home,
+   * Note: technically $XDG_HOME_DATA could point outside the homedir, but we ignore that.
+   */
+  return TRUE;
+}
+
 static void
 flatpak_context_take_filesystem (FlatpakContext        *context,
                                  char                  *fs,
-                                 FlatpakFilesystemMode  mode)
+                                 FlatpakFilesystemMode  mode,
+                                 FlatpakFilesystemFlags flags)
 {
+  if (mode > FLATPAK_FILESYSTEM_MODE_NONE)
+    {
+      g_assert ((flags & FLATPAK_FILESYSTEM_FLAGS_RESET) == 0);
+
+      if (strcmp (fs, "host") == 0)
+        context->fs_host_flags &= ~FLATPAK_FILESYSTEM_FLAGS_RESET;
+      else if (strcmp (fs, "home") == 0)
+        context->fs_home_flags &= ~FLATPAK_FILESYSTEM_FLAGS_RESET;
+    }
+  else if (flags & FLATPAK_FILESYSTEM_FLAGS_RESET)
+    {
+      if (strcmp (fs, "host") == 0)
+        context->fs_host_flags |= FLATPAK_FILESYSTEM_FLAGS_RESET;
```

Comment:
This seems to do nothing in the case someone does --nofilesystem=host but the reset flag was already set. Whereas in my implementation it would override that back non-reset. Its not immediately obvious what is right, but i *think* removing the reset flag is right.

Reply:
This is pretty subtle! I think the only time this would matter is if we have `--nofilesystem=host:reset --nofilesystem=host`, in that order, at the *same* level of precedence (not a different level of precedence, like manifest < overrides < CLI). Do you agree?

I think it maybe makes sense to err on the side of more protection/less exposure here: `--nofilesystem=host` doesn't really explicitly say "enable inheritance from lower layers", it just doesn't say "disable inheritance from lower layers" either. If we had an explicit `:no-reset` syntax then of course we'd want that to re-enable inheritance, but we don't, and I don't really see a reason why we would want that?

Reply:
I agree that this only applies on the same level, so multiple CLI args or multiple listing in a filesystem=... in an override file.

I can buy your argument for the current behavior, but we should add  a comment about that in the code.

Reply:
I added a comment.